### PR TITLE
Remove obsolete `http.status_text` from semantic conventions

### LINKED ISSUE #950: Remove semantic convention for http.status_text
[HTTP semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md) defines `http.status_text` as [`HTTP reason phrase`](https://tools.ietf.org/html/rfc7230#section-3.1.2). But even RFC sounds like that reason phrase is an obsolete artefact (emphasis is mine):

> The reason-phrase element exists for the sole purpose of providing a
   textual description associated with the numeric status code, **mostly
   out of deference to earlier Internet application** protocols that were
   more frequently used with interactive text clients.  A **client SHOULD
   ignore** the reason-phrase content

Should we drop this semantic attribute as serving no actual purpose?

### PR DESCRIPTION
**Pull request by arminru**
Fixes #950.

### CODE REVIEW

File: specification/trace/semantic_conventions/http.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ In the case of an HTTP redirect, the request should normally be considered succe
 unless the client aborts following redirects due to hitting some limit (redirect loop).
 If following a (chain of) redirect(s) successfully, the status should be set according to the result of the final HTTP request.
 
-Don't set the span status description if the reason can be inferred from `http.status_code` and `http.status_text`.
+Don't set the span status description if the reason can be inferred from `http.status_code`.
```
Comment (Oberon00) [inline specification/trace/semantic_conventions/http.md:47]: I think we can leave this out completely.

**Review by Oberon00 — APPROVED** [review]

**Review by iNikem — APPROVED** [review]

File: specification/trace/semantic_conventions/http.md (hunk 1 above)
Comment (arminru) [reply specification/trace/semantic_conventions/http.md:47]: I think it's fine to mention that one should not set the reason phrase as status message.

**Review by cijothomas — APPROVED** [review]

**Comment by justinfoote** [comment]
I'm totally in favor of this change.  Can we remove it from the [HTTP metrics semantic convention](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/metrics/semantic_conventions/http-metrics.md#labels) at the same time, or should I open a separate issue for that?

**Comment by Oberon00** [comment]
I think we should remove it in this PR actually.

**Comment by arminru** [comment]
@justinfoote I've already done that, see https://github.com/open-telemetry/opentelemetry-specification/pull/972/files#diff-0a939af5f14b4209f713354478b76841. 🙂

**Comment by justinfoote** [comment]
😅 Sorry I missed that.  Looks good to me then.

**Review by tigrannajaryan — APPROVED** [review]

**Comment by arminru** [comment]
Sufficient approvals on here and wide consensus on the issue to remove it, merging.

**Pull request merged**